### PR TITLE
Implement buyer able to leave feedback on products

### LINKED ISSUE #105: A buyer should be able to leave feedback for products
### Description

This feature will allow buyers to leave feedback on products they have purchased, which will help future buyers to make informed decisions. This feedback will be visible to all users who access the product page, including buyers, vendors, and admins. Buyers will have the capability to create, update, and delete their feedback. Admins will have the ability to view and delete any feedback, while vendors will only be able to view the feedback.

### Functionalities

- Buyers can leave feedback on products they have purchased.
- Buyers can manage the feedback by updating and deleting them.
- All users that view product page.
- Admin management can view and delete any feedback.

### Acceptance Criteria 

- A logged-in buyer can submit feedback on a product.
- Buyers can update or delete their own feedback.
- Feedback is visible to all users who access the product page.
- Admins can view and delete any feedback and Vendor can only view the product.

### Definition of Done

- [ ] Backend API endpoints for creating, updating, viewing, and deleting feedback are implemented and functional.
- [ ] Validation is in place to ensure only logged-in buyers can leave, update, or delete their feedback.
- [ ] Admin controls for managing feedback are implemented.
- [ ] Tests are practiced to ensure the system works correctly and is easy to use.
- [ ] User feedback is collected and used to enhance the feedback system.

### PR DESCRIPTION
**Pull request by UwicyezaG**
### Description

This feature will allow buyers to leave feedback on products they have purchased, which will help future buyers to make informed decisions. This feedback will be visible to all users who access the product page, including buyers, vendors, and admins. Buyers will have the capability to create, update, and delete their feedback. Admins will have the ability to view and delete any feedback, while vendors will only be able to view the feedback.

### Functionalities

- Buyers can leave feedback on products they have purchased.
- Buyers can manage the feedback by updating and deleting them.
- All users that view product page.
- Admin management can view and delete any feedback.

### Acceptance Criteria 

- A logged-in buyer can submit feedback on a product.
- Buyers can update or delete their own feedback.
- Feedback is visible to all users who access the product page.
- Admins can view and delete any feedback and Vendor can only view the product.

### Definition of Done

- [ ] Backend API endpoints for creating, updating, viewing, and deleting feedback are implemented and functional.
- [ ] Validation is in place to ensure only logged-in buyers can leave, update, or delete their feedback.
- [ ] Admin controls for managing feedback are implemented.
- [ ] Tests are practiced to ensure the system works correctly and is easy to use.
- [ ] User feedback is collected and used to enhance the feedback system.

### CODE REVIEW

**Review by elijahladdie — APPROVED** [review]

**Review by elijahladdie — APPROVED** [review]
LGTM

File: src/services/feedbackServices/createFeedback.ts
Comment (faid-terence) [inline src/services/feedbackServices/createFeedback.ts:1]: @UwicyezaG and @elijahladdie 
Check if a buyer has bought that product !

**Review by faid-terence — CHANGES_REQUESTED** [review]

File: src/services/feedbackServices/createFeedback.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    const product = await productRepository.findOne({ where: { id: productId } });
+    if (!product) {
+      return responseError(res, 404, `Your feedback can't be recorded at this time product not found`);
+    }
+    const order = await orderRepository.findBy({ id: orderId, orderStatus: 'completed', buyer: { id: req.user?.id } })
+    if (!order) {
+      return responseError(res, 404, `Your feedback can't be recorded at this time Your order doesn't exist or it haven't been completed yet`);
+    }
+
```
Comment (elijahladdie) [inline src/services/feedbackServices/createFeedback.ts]: @faid-terence You mean something like this

**Review by elijahladdie — APPROVED** [review]

File: src/services/feedbackServices/createFeedback.ts
Comment (faid-terence) [reply src/services/feedbackServices/createFeedback.ts:1]: @UwicyezaG and @elijahladdie 
Remove console.logs

**Review by faid-terence — CHANGES_REQUESTED** [review]

File: src/services/feedbackServices/createFeedback.ts
Comment (elijahladdie) [reply src/services/feedbackServices/createFeedback.ts:1]: Console already removed 🙂

**Review by faid-terence — APPROVED** [review]
Nice Work @UwicyezaG and @elijahladdie !
PR Approved !

**Pull request merged**